python/plugins/processing/algs/gdal/gdalcalc.py

```diff
@@ -131,38 +206,35 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         #if debug:
         #    arguments.append('--debug')
         arguments.append('-A')
-        arguments.append(self.getParameterValue(self.INPUT_A))
-        if self.getParameterValue(self.BAND_A):
-            arguments.append('--A_band ' + self.getParameterValue(self.BAND_A))
-        if self.getParameterValue(self.INPUT_B):
+        arguments.append(self.parameterAsLayer(parameters, self.INPUT_A, context).source())
```

Comment:
This needs a bit more safety:

    layer_a = self.parameterAsRasterLayer(parameters, self.INPUT_A, context)
    if layer_a is None:
            raise QgsProcessingException(self.invalidRasterError(parameters, self.INPUT_A))
    arguments.append(layer_a.source())